README.md

```diff
@@ -1,3 +1,166 @@
 # amqp
 
-Simplified publish/subscribe and connection retry handling for amqp (rabbitmq)
+Simplified publish/subscribe and connection retry handling for amqp (rabbitmq).
+
+This package integrates with [amqplib](http://www.squaremobius.net/amqp.node/channel_api.html) and [amqp-connection-manager](https://github.com/benbria/node-amqp-connection-manager) (which is also built on amqplib).  Use one of those packages to create connections that can be used with this package.
+
+## Install
+
+```
+npm install @nowait/amqp --save
+```
+
+## Examples
+
+### Create a Channel
+
+First, create an AMQP connection, using either amqplib or amqp-connection-manager.  Then, create a channel.
+
+#### Create duplex channel
+
+Duplex channels are provided by amqplib, and can be used for either publishing or consuming messages.  They don't provide any fault tolerance for publishers: when connections are dropped, publishers will fail.
+
+```js
+import amqp from 'amqplib'
+import { createChannel } from '@nowait/amqp'
+
+// See amqplib documentation
+// http://www.squaremobius.net/amqp.node/channel_api.html#connect
+const connection = amqp.connect(uri, options)
+
+const channel = createChannel(connection)
+```
+
+#### Create managed channel
+
+Managed channels can only be used for event *publishers*.  They provide fault-tolerance by buffering events when the connection is dropped.  It will attempt to reconnect, and then deliver the buffered events.
+
+```js
+import amqpcm from 'amqp-connection-manager'
+import { createManagedChannel } from '@nowait/amqp'
+
+// See amqp-connection-mananger documentation
+// https://github.com/benbria/node-amqp-connection-manager#basics
+const connection = amqpcm.connect([...uri], options)
+
+const channel = createManagedChannel(connection)
+```
+
+### Publish messages to a queue
+
+After creating a Channel, use `publishTo` to publish messages to a queue.
+
+```js
+// Assuming channel was created using one of the approaches above
+
+import { publishTo } from '@nowait/amqp'
+
+const publishChannel = publishTo(channel)
+
+const publishConfig = {
+  exchangeName: 'some-exchange',
+  routingKey: 'some-routing-key'
+}
+
+// Create a publisher function that will publish messages
+// using the exchangeName, queueName, and routingKey
+const publisher = publishChannel(queueConfig)
```

Comment:
I was kinda using "publish" like an adjective, i.e. "a publish channel".  I know that's not really english, tho :) Do you think it'd be better for `publishChannel` to be named as a verb, like `createPublisherFunction`?